lib/ecto_cursor_based_stream.ex

```diff
@@ -93,31 +137,116 @@ defmodule EctoCursorBasedStream do
 
   defp parse_options(options) do
     max_rows = Keyword.get(options, :max_rows, 500)
-    after_cursor = Keyword.get(options, :after_cursor)
+    after_cursor = Keyword.get(options, :after_cursor, nil)
     cursor_field = Keyword.get(options, :cursor_field, :id)
+    order = Keyword.get(options, :order, :asc)
+
+    repo_opts =
+      Keyword.take(options, [:prefix, :timeout, :log, :telemetry_event, :telemetry_options])
+
+    cursor_fields = validate_cursor_fields(cursor_field)
 
     %{
       max_rows: max_rows,
-      after_cursor: after_cursor,
-      cursor_field: cursor_field
+      cursor_fields: cursor_fields,
+      after_cursor: validate_initial_cursor(cursor_fields, after_cursor),
+      order: order,
+      repo_opts: repo_opts
     }
   end
 
-  defp get_rows(repo, query, cursor_field, cursor, max_rows) do
+  defp validate_cursor_fields(value) do
+    cursor_fields = List.wrap(value)
+
+    if Enum.all?(cursor_fields, &is_atom/1) do
+      cursor_fields
+    else
+      raise ArgumentError,
+            "EctoCursorBasedStream expected `cursor_field` to be an atom or list of atoms, got: #{inspect(value)}."
+    end
+  end
+
+  defp validate_initial_cursor(_, nil) do
+    nil
+  end
+
+  defp validate_initial_cursor(cursor_fields, %{} = after_cursor) do
+    {after_cursor, rest} = Map.split(after_cursor, cursor_fields)
+
+    if map_size(rest) == 0 do
+      after_cursor
+    else
+      raise ArgumentError,
+            "EctoCursorBasedStream expected `after_cursor` to be a map with fields #{inspect(cursor_fields)}, got: #{inspect(rest)}."
+    end
+  end
+
+  defp validate_initial_cursor([cursor_field], after_cursor) when not is_list(after_cursor) do
+    %{cursor_field => after_cursor}
+  end
+
+  defp validate_initial_cursor(cursor_fields, after_cursor) do
+    raise ArgumentError,
+          "EctoCursorBasedStream expected `after_cursor` to be a map with fields #{inspect(cursor_fields)}, got: #{inspect(after_cursor)}."
```

Comment:
It actually expects after_cursor to be a map or **integer/string** and btw why not other types which we can directly compare using <, > in Ecto?